pkg/controller/jobs/job/job_multikueue_adapter.go

```diff
@@ -38,12 +38,33 @@ import (
 	"sigs.k8s.io/kueue/pkg/features"
 	"sigs.k8s.io/kueue/pkg/util/api"
 	clientutil "sigs.k8s.io/kueue/pkg/util/client"
+	"sigs.k8s.io/kueue/pkg/workloadslicing"
 )
 
 type multiKueueAdapter struct{}
 
 var _ jobframework.MultiKueueAdapter = (*multiKueueAdapter)(nil)
 
+// updateLabel helper sets the provided label key/value on the provided job.
+// Returns true, if the job's label was changed.
+func updateLabel(job *batchv1.Job, labelKey, labelValue string) bool {
+	labels := job.Labels
+	if labels == nil {
+		job.Labels = map[string]string{labelKey: labelValue}
+		return true
+	}
+	value, found := labels[labelKey]
+	labels[labelKey] = labelValue
+	return !found || value != labelValue
+}
+
```

Comment:
```suggestion
```

I think that we can rely on `utilmaps.Copy()` function. See the following comment.